Arrowgene.Ddon.GameServer/Enemy/EnemyManager.cs

```diff
@@ -27,50 +29,13 @@ public List<EnemySpawn> GetSpawns(CStageLayoutID stageLayoutId, byte subGroupId)
 
         public List<EnemySpawn> GetSpawns(StageId stageId, byte subGroupId)
         {
-            if (!_spawns.ContainsKey(stageId))
-            {
-                return new List<EnemySpawn>();
-            }
+            // This is probably a terrible idea but i adore the syntax
+            var spawns = 
+                from spawn in _assetRepository.EnemySpawns
+                where spawn.StageId.Equals(stageId) && spawn.SubGroupId == subGroupId
+                select spawn;
 
-            Dictionary<byte, List<EnemySpawn>> stageSpawns = _spawns[stageId];
-            if (!stageSpawns.ContainsKey(subGroupId))
-            {
-                return new List<EnemySpawn>();
-            }
-
-            // TODO check time of day
-
-            return new List<EnemySpawn>(stageSpawns[subGroupId]);
-        }
-
-        public void Load()
```

Comment:
I would prefer to have a Load() method, or a way to tell the manager to reload its state in a controlled way.
This way a reload could potentially be triggered from a chat command.

I also do like the idea of your file watcher, I'm thinking that it could trigger the Load() method as well then. (if the asset repository exposes a `File changed`-event then the EnemyManager could subscribe to that and act on it

Lastly if we decide to keep the load method, it should potentially clear the `_spawns` at the beginning too

Reply:
I've thought about the asset repository emitting events however I'm unsure on how to implement it on C#, mind giving me some pointers to look into it?